File: leetcode/2760.最长奇偶子数组.py
class Solution:
    def longestAlternatingSubarray(self, nums: List[int], threshold: int) -> int:
        i, n = 0, len(nums)
        ans = 0
        while i < n:
            if nums[i] % 2 != 0 or nums[i] > threshold:
                i += 1
                continue
            start = i
            i += 1
            while i < n and nums[i] % 2 != nums[i-1] % 2 and nums[i] <= threshold:
                i += 1
            ans = max(ans, i - start)
        return ans

File: leetcode/2760.最长奇偶子数组.go
func longestAlternatingSubarray(nums []int, threshold int) int {
    //分组循环模板题目
    n := len(nums)
    i := 0
    ans := 0
    for i < n {
        if nums[i] % 2 != 0 || nums[i] > threshold {
            i += 1
            continue   
        }    
        start := i
        i += 1
        for i < n && nums[i] % 2 != nums[i - 1] % 2 && nums[i] <= threshold {
            i += 1
        }
        ans = max(ans, i - start)
    }
    return ans
}